biom/table.py

```diff
@@ -2935,6 +2921,161 @@ def _intersect_id_order(self, a, b):
                 idx += 1
         return new_order
 
+    def concat(self, others, axis='sample'):
```

Comment:
After reading the code in here I think it makes sense to break it in a couple of private functions and probably upgrade this to a class method, it will be easier to do this change now than later and break compatibility on the API.